src/Doctrine/Orm/State/ItemProvider.php

```diff
@@ -53,8 +53,11 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             $entityClass = $options->getEntityClass();
         }
 
-        /** @var EntityManagerInterface $manager */
+        /** @var EntityManagerInterface|null $manager */
         $manager = $this->managerRegistry->getManagerForClass($entityClass);
+        if (null === $manager) {
+            throw new RuntimeException(\sprintf('No manager found for class "%s". Are you sure it\'s an entity?', $entityClass));
```

Comment:
As this exceptional case relates to what the developer wrote; should this not be considered a `LogicException` instead?

Reply:
I don't think so as this may be that the manager isn't properly configured at runtime.

Reply:
I agree with @rvanlaak 
I don't see how the manager could not be properly configured... But not a big deal imho.